src/tasks/wifi.c

```diff
@@ -491,10 +518,104 @@ static void do_beacon()
         };
         send_beacon(serial, ips);
 
-        /* Now flush all incomming data since we don't care about it. */
+
+        /* Now flush all incoming data since we don't care about it. */
         serial_purge_rx_queue(serial);
+
+do_beacon_exit:
+        xSemaphoreGive(state.connection_mutex);
+}
+
+static bool start = false;
+
+
+/* TODO this should get broken out to a modular design */
+
+/* Camera control for Hero 4+, including session */
+/*
+static void send_camera_control(struct Serial* serial, const char* ips[])
+{
+    if (start) {
+            serial_write_s(serial, "GET /gp/gpControl/command/shutter?p=1 HTTP/1.0\r\n\r\n");
+    }
+    else {
+            serial_write_s(serial, "GET /gp/gpControl/command/shutter?p=0 HTTP/1.0\r\n\r\n");
+    }
+    start = !start;
+}
+*/
+
+/* Camera control for Hero 2,3
+ * Format is: GET /bacpac/SH?t=<wifi-password>&p=<shutter-param>
+ */
+static void send_camera_control(struct Serial* serial)
+{
+    const char preamble[] = "GET /bacpac/SH?t=";
```

Comment:
Make static so its not allocated every time.  Same with others that don't change.

Reply:
Err... not allocated.  I mean not populated every invocation.  Also will save on stack space.